index.js

```diff
@@ -109,6 +109,10 @@ export default function pThrottle({limit, interval, strict, onDelay}) {
 			},
 		});
 
+		if (signal) {
+			signal.addEventListener('abort', throttled.abort);
+		}
```

Comment:
You also need to use `AbortSignal.throwIfAborted()` as early as possible in the function in case the signal was already aborted when this is called.